### PR TITLE
feat: let pylance use sub-level logger of logging

### LINKED ISSUE #3195: Pylance shoud use a sub-level logger of logging
Currently, I noticed that PyLance directly uses logging to record logs. So, when we need to modify the log level, we have to do it in the following way:

```
logging.basicConfig(level=logging.DEBUG)
```

The intention behind this is to achieve the goal by modifying the root logger of logging. However, this doesn't seem to be an optimal practice. The optimal practice should be to use the sublevel of logging to prevent changes to the log level from having out-of-scope impacts, [like the fsspec's practice](https://github.com/fsspec/filesystem_spec/blob/master/fsspec/utils.py#L480). Generally, in a typical Python project, multiple frameworks are usually used, and they may all use the logging modules. If we modify the log level of the root logger, the impact scope of this change may exceed expectations (or there may be side effects).
Therefore, my suggestion is to use the sub-logger and make the following changes:

* Use `logging.getLogger()` to obtain pylance's logger;
* Use APIs or environment variables to control the log level of the PyLance logger.

### PR DESCRIPTION
**Pull request by yanghua**
Closes #3195

### CODE REVIEW

**Comment by yanghua** [comment]
cc @westonpace @wjones127 Please have a look, thanks.

File: python/python/lance/log.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import os
+from typing import Optional
+
+ENV_NAME_PYLANCE_LOGGING_LEVEL = "PYLANCE_LOGGING_LEVEL"
+
+logger = logging.getLogger("pylance")
+logger.setLevel(os.environ.get(ENV_NAME_PYLANCE_LOGGING_LEVEL, "INFO"))
+
+
+def set_logger(
```
Comment (yanghua) [inline python/python/lance/log.py:14]: Here, we open the ability for users to configure the pylance's logger to align with themself.

**Review by wjones127 — COMMENTED** [review]
This seems alright. Though most of our logging happens at the Rust layer. That does make me wonder if it is even that valuable to let users tweak the logger, given it only controls a small fraction of our logs.

I made a suggestion on how they could share configuration. Otherwise, I think this seems fine.

File: python/python/lance/log.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    if lh is None:
+        lh = logging.FileHandler(file_path)
+    lh.setLevel(level)
+    formatter = logging.Formatter(format_string)
+    lh.setFormatter(formatter)
+    logger.addHandler(lh)
+
+
+def get_logger():
+    return logger
```
[the remark this answers is not in the document]
Comment (yanghua) [reply python/python/lance/log.py:36]: Make sense, accept.

File: python/python/lance/log.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+
+import logging
+import os
+from typing import Optional
+
+ENV_NAME_PYLANCE_LOGGING_LEVEL = "PYLANCE_LOGGING_LEVEL"
```
[the remark this answers is not in the document]
Comment (yanghua) [reply python/python/lance/log.py:8]: Thanks for your explanation. Reasonable. Accept.

By the way, it seems there is no description about the env `LANCE_LOG ` in the whole codebase(not sure about documentation in the official site). 

If you agree, we should document it for users. WDYT?

**Review by wjones127 — APPROVED** [review]
Nice work. And agreed, we should add some documentation about the logging settings.

**Pull request merged**